libp2p/transports/tcptransport.nim

```diff
@@ -85,7 +137,7 @@ method dial*(t: TcpTransport,
   trace "dialing remote peer", address = $address
   ## dial a peer
   let client: StreamTransport = await connect(address)
-  result = await t.connHandler(t.server, client, true)
+  result = t.connHandler(t.server, client, true)
```

Comment:
I think this needs to be `async`, returning the future will lead to issues when mixing with exceptions, as we've seen in other places. In fact, this is the same issue discussed in the next comments block bellow.

Reply:
if you await that, it will block there, while we want the lvalue receiver await ( I think that's the logic, I'll check it again )

Reply:
see above for `connHandler` changes, basically here we want lvalue side awaiting